### PR TITLE
Makes the 'research mode' of radiation collectors and tesla coins generate discovery research instead of generic research.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes the 'research mode' of radiation collectors and tesla coins generate discovery research instead of generic research.

## Why It's Good For The Game
It's a very rare even when Engineering decides to use the machinery described above for research purposes. Building a tesla requires time, space, credits and approval from CE/Captain. And doing research with the rad collectors requires the gas tanks to be filled with O2 and Tritium. 
Imagine the disappointment of the engineers when they find out that all their hard work has produced glorified R&D servers. This PR alleviates that while promoting cross-department interactions between Science, Engineering and Cargo.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Rad collectors : 

https://user-images.githubusercontent.com/110184118/208490066-2fe91c50-7690-43c5-ad7c-22ff4c4ce3dc.mp4

Tesla Coils : 

https://user-images.githubusercontent.com/110184118/208490093-d881955f-3796-4f7f-8ab9-737996748501.mp4

</details>

## Changelog
:cl:
tweak: Made the Rad Collector and the Tesla coil generate discovery points alongside general research points while in research mode.
tweak: Increased the amount of money added to the Engineering budget to the Tesla coils in research mode.
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
Now, I *might* have made them produce too much? But it's such a rare event that I think it's fine, and there's no way this can overshadow a research TTV, a miner rushing to scan megafauna or explorers stumbling on a clockwork ruin on their first mission in terms on discovery point turnout.

**Comment by EvilDragonfiend** [comment]
I think It should have max value how much they can get from those

**Comment by EvilDragonfiend** [comment]
my suggestion is
* it has a default increasing value
* the severity determines its maximum

**Comment by CydiaLamiales** [comment]
> the severity determines its maximum

Severity of what? Also I'm against limiting it, afterall you need to make absurdly dangerous engines to reach absurdly high values.

**Comment by Fronsis** [comment]
> > the severity determines its maximum
> 
> Severity of what? Also I'm against limiting it, afterall you need to make absurdly dangerous engines to reach absurdly high values. And remember, a maxcap TTV can easily be made in 8 minutes from roundstart and nets you all the discovery points you'll ever need, while this could take 20+ minutes to set up.

Agreed, if this is capped properly this could mean that low-pop shifts could be saved tech-wise because if you got a competent engineer that knows how to use rad collectors and set up the SM they could use them to help slowly getting Discovery research aiding a lot in the lack of maxcap/miners/explorers on the shift preventing being with tier 2 one hour and a half into the round.

**Review by itsmeow — DISMISSED** [review]
I like this, although I'd like to see how it works ingame. I have no problem with merging it though since it's an easy tweak should it be too high.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Make it generate both

**Comment by CydiaLamiales** [comment]
> Make it generate both

Alright, but uh, can I increase the amount of dosh added to the engineering budget too? It's comically small

File: code/modules/power/tesla/coil.dm [hunk 1]
```diff
@@ -130,9 +130,10 @@
 		tesla_zap(src, 5, power_produced, tesla_flags, shocked_things)
 		var/datum/bank_account/D = SSeconomy.get_budget_account(ACCOUNT_ENG_ID)
 		if(D)
-			D.adjust_money(min(power_produced, 3))
+			D.adjust_money(min(power_produced, 3)*10)
 		if(istype(linked_techweb))
-			linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, min(power_produced, 3)) // x4 coils with a pulse per second or so = ~720/m point bonus for R&D
+			linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, min(power_produced, 3)*10)
+			linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DISCOVERY, min(power_produced, 3)*10) // x4 coils with a pulse per second or so = ~7200/m point bonus for R&D
```
Comment (PowerfulBacon) [inline code/modules/power/tesla/coil.dm]: 7200 points per minute? Thats insane (For money too, 7200 a minute is just ridiculous)

Comment (CydiaLamiales) [reply code/modules/power/tesla/coil.dm]: Alright, I'll turn those down. By how much though?

Comment (itsmeow) [reply code/modules/power/tesla/coil.dm]: It should be like 400

Comment (CydiaLamiales) [reply code/modules/power/tesla/coil.dm]: Too little, it was 720 before and that was slow as hell.

Comment (CydiaLamiales) [reply code/modules/power/tesla/coil.dm]: There, I adjusted the values and uploaded new videos with them

**Review by Tsar-Salat — COMMENTED** [review]
Yeah, looks good. Considering the state of Science and research bombs, more methods for other departments to generate points is a good addition. There really is low benefit to cost for tesla and singulos right now.

**Review by PowerfulBacon — APPROVED** [review]
There is a 50% chacne this is absolutely overpowered and produces way too many research points but I've never used them so I cba to do the math

**Pull request merged**